q2cli/tests/test_tools.py

```diff
@@ -724,5 +725,154 @@ def test_multiple_file_tsv(self):
+
+    def test_show_types_tsv(self):
+        result = self.runner.invoke(tools, ['show-types', '--tsv'])
+        self.assertEqual(result.exit_code, 0)
+
+        # len - 1 because \n split produces a final ''
+        self.assertEqual(len(result.output.split('\n')) - 1,
+                         len(self.pm.artifact_classes))
+
+        no_description_count = 0
+        for name, artifact_class_record in self.pm.artifact_classes.items():
+            self.assertIn(name, result.output)
+            self.assertIn(artifact_class_record.description, result.output)
+            if artifact_class_record.description == '':
+                no_description_count += 1
+
+        self.assertEqual(no_description_count, result.output.count('\t\n'))
+
+
+class TestShowFormats(unittest.TestCase):
+    def setUp(self):
+        self.runner = CliRunner()
+        self.pm = PluginManager()
+
+    def tearDown(self):
+        pass
+
+    def test_show_all_importable_formats(self):
+        result = self.runner.invoke(tools, ['show-formats', '--importable'])
+        self.assertEqual(result.exit_code, 0)
+
+        for name, format_record in self.pm.importable_formats.items():
+            self.assertIn(name, result.output)
+            docstring = format_record.format.__doc__
+            if docstring:
+                description = docstring.split('\n\n')[0].strip()
+                for word in description:
+                    self.assertIn(word.strip(), result.output)
+
+    def test_show_all_exportable_formats(self):
+        result = self.runner.invoke(tools, ['show-formats', '--exportable'])
+        self.assertEqual(result.exit_code, 0)
+
+        for name, format_record in self.pm.exportable_formats.items():
+            self.assertIn(name, result.output)
+            docstring = format_record.format.__doc__
+            if docstring:
+                description = docstring.split('\n\n')[0].strip()
+                for word in description:
+                    self.assertIn(word.strip(), result.output)
+
+    def test_show_formats_fuzzy(self):
+        formats = list(self.pm.importable_formats)[:5]
+        result = self.runner.invoke(tools, ['show-formats', '--importable',
+                                            *formats])
+        self.assertEqual(result.exit_code, 0)
+
+        # see TestShowTypes.test_show_types_fuzzy
+        self.assertGreaterEqual(len(result.output.split('\n\n')) - 1,
+                                len(formats))
+
+    def test_show_formats_strict(self):
+        formats = list(self.pm.exportable_formats)[:5]
+        result = self.runner.invoke(tools, ['show-formats', '--exportable',
+                                            '--strict', *formats])
+        self.assertEqual(result.exit_code, 0)
+        print(formats)
+        print(result.output)
+        self.assertEqual(len(result.output.split('\n\n')) - 1, len(formats))
+
+        result = self.runner.invoke(tools, ['show-formats', '--exportable',
```

Comment:
I think this actually isn't testing that `strict` is working here, because these queries wouldn't return a hit in either case: 

```bash
$ qiime tools show-formats --importable nonsense
$ qiime tools show-formats --importable nonsense --strict
$
```

I would instead create something that is a close match to an existing type, and then confirm that you get expected results here with and without `strict`, to test that toggling the parameter behaves as expected.